server/svix-server/src/core/security.rs

```diff
@@ -110,14 +239,47 @@ where
         .one(db)
         .await?
         .ok_or_else(|| HttpError::not_found(None, None))?;
+
+        if let Some(permitted_app_id) = &permissions.app_id {
+            if permitted_app_id != &app.id {
+                return Err(HttpError::unauthorized(
+                    None,
+                    Some(
+                        "You are only permitted to perform operations under the Application type"
+                            .to_owned(),
+                    ),
+                )
+                .into());
+            }
+        }
+
         Ok(AuthenticatedApplication { permissions, app })
     }
 }
 
-pub fn generate_token(keys: &Keys, org_id: OrganizationId) -> Result<String> {
-    let claims = Claims::create(Duration::from_hours(24 * 365 * 10))
-        .with_issuer(env!("CARGO_PKG_NAME"))
-        .with_subject(org_id.0);
+pub fn generate_org_token(keys: &Keys, org_id: OrganizationId) -> Result<String> {
+    let claims = Claims::with_custom_claims(
+        CustomClaim { organization: None },
+        Duration::from_hours(24 * 365 * 10),
+    )
+    .with_issuer(env!("CARGO_PKG_NAME"))
```

Comment:
Nit (since I already made other comments): have a constant variable called `JWT_ISSUER` that stores this.